doc/fiber.md

```diff
@@ -1,16 +1,46 @@
-# Scheduler
+# Fiber
+
+Fibers provide a mechanism for cooperative concurrency.
+
+## Context Switching
+
+Fibers execute a user provided block. During the execution, the block may call `Fiber.yield` or `Fiber.transfer` to switch to another fiber. `Fiber#resume` is used to continue execution from the point where `Fiber.yield` was called.
```

Comment:
^user provided^user-provided

Also, long lines are typically cut (I don't care personally)

Reply:
My editor does it for me :p

Reply:
Fixed.